python/search_completer/global_search_completer.py

```diff
@@ -0,0 +1,219 @@
+        # criterial by calling ``set_searchable_entity_types``.
+        self._entity_search_criteria = {
+            "Asset": [],
+            "Shot": [],
+            "Task": [],
+            "HumanUser": [["sg_status_list", "is", "act"]],
+            "Group": [],
+            "ClientUser": [["sg_status_list", "is", "act"]],
+            "ApiUser": [],
+            "Version": [],
+            "PublishedFile": [],
+        }
+
+    def get_result(self, model_index):
+        """
+        Return the entity data for the supplied model index or None if there is
+        no data for the supplied index.
+
+        :param model_index: The index of the model to return the result for.
+        :type model_index: :class:`~PySide.QtCore.QModelIndex`
+
+        :return: The entity dict for the supplied model index.
+        :rtype: :obj:`dict`: or ``None``
+        """
+
+        # make sure that the user selected an actual shotgun item.
+        # if they just selected the "no items found" or "loading please hold"
+        # items, just ignore it.
+        mode = shotgun_model.get_sanitized_data(model_index, self.MODE_ROLE)
+        if mode == self.MODE_RESULT:
+
+            # get the payload
+            data = shotgun_model.get_sanitized_data(model_index, self.SG_DATA_ROLE)
+
+            # Example of data stored in the data role:
+            #
+            # {'status': 'vwd',
+            #  'name': 'bunny_010_0050_comp_v001',
+            #  'links': ['Shot', 'bunny_010_0050'],
+            #  'image': 'https://xxx',
+            #  'project_id': 65,
+            #  'type': 'Version',
+            #  'id': 99}
+
+            # NOTE: this data format differs from what is typically returned by
+            # the shotgun python-api. this data may be formalized at some point
+            # but for now, only expose the minimum information.
+
+            return {
+                "type": data["type"],
+                "id": data["id"],
+                "name": data["name"],
+            }
+        else:
+            return None
+
+    def set_searchable_entity_types(self, types_dict):
+        """
+        Specify a dictionary of entity types with optional search filters to
+        limit the breadth of the widget's search.
+
+        Use this method to override the default searchable entity types
+        dictionary which looks like this::
+
+          {
+            "Asset": [],
+            "Shot": [],
+            "Task": [],
+            "HumanUser": [["sg_status_list", "is", "act"]],    # only active users
+            "Group": [],
+            "ClientUser": [["sg_status_list", "is", "act"]],   # only active users
+            "ApiUser": [],
+            "Version": [],
+            "PublishedFile": [],
+          }
+
+        :param types_dict: A dictionary of searchable types with optional filters
+
+        """
+        self._entity_search_criteria = types_dict
+
+    def _set_item_delegate(self, popup, text):
```

Comment:
These code under this line is new. Those methods are called by the base class, SearchCompleter